### PR TITLE
55 Exception Handling in UI

### PR DESCRIPTION
**Pull request by mac641**
- Add exception handling in UI by showing dialogs (ExceptionHandler.kt)
- Add ConfigManager usage throughout the whole app
- NotificationConfig
- Minor refactoring
- Edit detekt rules
- Add detekt formatter

### CODE REVIEW

File: app/src/main/java/de/hsaugsburg/teamulster/sohappy/fragment/SmileFragment.kt [hunk 1]
```diff
@@ -174,8 +175,7 @@ class SmileFragment : Fragment() {
                 if (nextTick > 0) {
                     binding.countdownText.animate()
                         .alpha(1f)
-                        .translationYBy(100f)
-                        .setDuration(125)
+                        .translationYBy(100f).duration = 125
```
Comment (limkokyew) [inline app/src/main/java/de/hsaugsburg/teamulster/sohappy/fragment/SmileFragment.kt:178]: Minor nitpick, but maybe move `.duration = 125` to a new line

File: app/src/main/java/de/hsaugsburg/teamulster/sohappy/fragment/SmileFragment.kt [hunk 2]
```diff
@@ -188,13 +188,11 @@ class SmileFragment : Fragment() {
         }
 
         binding.textView.animate()
-            .alpha(1f)
-            .setDuration(500)
+            .alpha(1f).duration = 500
```
Comment (limkokyew) [inline app/src/main/java/de/hsaugsburg/teamulster/sohappy/fragment/SmileFragment.kt:191]: See above

File: app/src/main/java/de/hsaugsburg/teamulster/sohappy/fragment/SmileFragment.kt [hunk 3]
```diff
@@ -188,13 +188,11 @@ class SmileFragment : Fragment() {
...
     }
 
     private fun fadeOutText() {
         binding.textView.animate()
-            .alpha(0f)
-            .setDuration(500)
+            .alpha(0f).duration = 500
```
Comment (limkokyew) [inline app/src/main/java/de/hsaugsburg/teamulster/sohappy/fragment/SmileFragment.kt:196]: See above

**Review by limkokyew — APPROVED** [review]
Good work! Just one minor suggestion for readability.

**Review by AnnaGsell — APPROVED** [review]

**Comment by mac641** [comment]
Waiting for #42 to be merged

**Pull request merged**